compiler/luci-interpreter/src/kernels/OneHot.test.cpp

```diff
@@ -0,0 +1,173 @@
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorShape(output_tensor), output_shape);
+  EXPECT_THAT(extractTensorData<T2>(output_tensor), ::testing::ElementsAreArray(output_data));
+}
+
+template <typename T> class OneHotTest : public ::testing::Test
+{
+};
+
+using DataTypes = ::testing::Types<float, uint8_t, int16_t>;
+TYPED_TEST_CASE(OneHotTest, DataTypes);
+
+TYPED_TEST(OneHotTest, BasicPattern)
+{
+  // axis 0
+  Check<int32_t, TypeParam>(/*input_shape=*/{2, 3}, /*output_shape=*/{4, 2, 3},
+                            /*input_data=*/
+                            {
+                              0, 3, 5, //
+                              7, 3, 0, //
+                            },
+                            /*depth_data=*/{4}, /*on_value_data=*/{1}, /*off_value_data=*/{0},
+                            /*axis=*/0,
+                            /*output_data=*/
+                            {
+                              1, 0, 0, //
+                              0, 0, 1, //
+
+                              0, 0, 0, //
+                              0, 0, 0, //
+
+                              0, 0, 0, //
+                              0, 0, 0, //
+
+                              0, 1, 0, //
+                              0, 1, 0, //
+                            });
+  // axis 1
+  Check<int32_t, TypeParam>(/*input_shape=*/{2, 3}, /*output_shape=*/{2, 4, 3},
+                            /*input_data=*/
+                            {
+                              0, 3, 5, //
+                              7, 3, 0, //
+                            },
+                            /*depth_data=*/{4}, /*on_value_data=*/{1}, /*off_value_data=*/{0},
+                            /*axis=*/1,
+                            /*output_data=*/
+                            {
+                              1, 0, 0, //
+                              0, 0, 0, //
+                              0, 0, 0, //
+                              0, 1, 0, //
+
+                              0, 0, 1, //
+                              0, 0, 0, //
+                              0, 0, 0, //
+                              0, 1, 0, //
+                            });
+  // axis -1
+  Check<int32_t, TypeParam>(/*input_shape=*/{2, 3}, /*output_shape=*/{2, 3, 4},
+                            /*input_data=*/
+                            {
+                              0, 3, 5, //
+                              7, 3, 0, //
+                            },
+                            /*depth_data=*/{4}, /*on_value_data=*/{1}, /*off_value_data=*/{0},
+                            /*axis=*/-1,
+                            /*output_data=*/
+                            {
+                              1, 0, 0, 0, //
+                              0, 0, 0, 1, //
+                              0, 0, 0, 0, //
+
+                              0, 0, 0, 0, //
+                              0, 0, 0, 1, //
+                              1, 0, 0, 0, //
+                            });
+}
```

Comment:
For reviewer: to define correct values for value check I used following script:

```python
import tensorflow as tf
import numpy as np

print("Enter axis:")
axis = int(input())
if axis < -1 or axis > 2:
    raise ValueError("Wrong axis is given")

with tf.Graph().as_default() as graph:
    # create model
    input_shape = (2, 3)
    ifm = tf.compat.v1.placeholder(tf.int32, shape = input_shape)
    ofm = tf.one_hot(ifm, depth=4, axis=axis)

    # invoke and save input ans output
    with tf.compat.v1.Session() as sess:
        in0 = np.array([0, 3, 5, 7, 3, 0]).reshape(input_shape).astype(np.int32)
        out0 = sess.run(ofm, {ifm: in0})

        # print result
        print("Input:")
        print(in0)
        print("Output shape:", out0.shape)
        print("Output data:")
        print(out0)
```

Results of launching:

1. axis `0`:
```
Enter axis:
0
Input:
[[0 3 5]
 [7 3 0]]
Output shape: (4, 2, 3)
Output data:
[[[1. 0. 0.]
  [0. 0. 1.]]

 [[0. 0. 0.]
  [0. 0. 0.]]

 [[0. 0. 0.]
  [0. 0. 0.]]

 [[0. 1. 0.]
  [0. 1. 0.]]]
```

2. axis `1`:
```
Enter axis:
1
Input:
[[0 3 5]
 [7 3 0]]
Output shape: (2, 4, 3)
Output data:
[[[1. 0. 0.]
  [0. 0. 0.]
  [0. 0. 0.]
  [0. 1. 0.]]

 [[0. 0. 1.]
  [0. 0. 0.]
  [0. 0. 0.]
  [0. 1. 0.]]]
```

3. axis `-1`:
```
Enter axis:
-1
Input:
[[0 3 5]
 [7 3 0]]
Output shape: (2, 3, 4)
Output data:
[[[1. 0. 0. 0.]
  [0. 0. 0. 1.]
  [0. 0. 0. 0.]]

 [[0. 0. 0. 0.]
  [0. 0. 0. 1.]
  [1. 0. 0. 0.]]]
```

_Originally posted by @m-bronnikov in https://github.com/Samsung/ONE/pull/8084#r761919939_